### DISCUSSION TITLE
ODF does not load releases

### DISCUSSION THREAD

**Discussion by mveccia** [GrandOrgue/grandorgue#1337] [Help] [answered]
in my odf I think I have added the lines needed to get the releases, but if I launch the .organ file it tells me that the various items are unused... why? 
code for a note:
Pipe001Gain=0
Pipe001HarmonicNumber=4
Pipe001TrackerDelay=0
Pipe001IsTremulant=0
Pipe001=Data\Pedalboard\Horn 4\A0\036-c.wav
Pipe001LoadRelease=No
Pipe001PitchTuning=0.00000
Pipe001ReleaseCount=4
Pipe001Release001=Data\Pedalboard\Horn 4\R0\036-c.wav
Pipe001Release001.MaxKeyPressTime=-1
Pipe001Release001.IsTremulant=0
Pipe001Release002=Data\Pedalboard\Horn 4\R1\036-c.wav
Pipe001Release002.MaxKeyPressTime=200
Pipe001Release002.IsTremulant=0
Pipe001Release003=Data\Pedalboard\Horn 4\R2\036-c.wav
Pipe001Release003.MaxKeyPressTime=300
Pipe001Release003.IsTremulant=0
Pipe001Release004=Data\Pedalboard\Horn 4\R3\036-c.wav
Pipe001Release004.MaxKeyPressTime=600
Pipe001Release004.IsTremulant=0
error response:
Warning: Unused ODF entry 'Rank001/Pipe001Release001.MaxKeyPressTime'
Warning: Unused ODF entry 'Rank001/Pipe001Release001.IsTremulant'

**Comment by eturpault** [comment, marked as answer]
It is because you added a point after Pipe001Release001, the right syntax is Pipe001Release001MaxKeyPressTime and Pipe001Release001IsTremulant

**Reply by mveccia** [reply]
That was it, thanks for the kind reply. I'm left with one last error: "Warning: Invalid integer value in section 'Stop001' entry 'FirstAccessiblePipeLogicalKeyNumber': 1...." for all stops. Do you know how it is calculated?

**Reply by eturpault** [reply]
The attribute 'FirstAccessiblePipeLogicalKeyNumber' indicates the key number, in the manual to which the stop is assigned, which will control the first pipe of the rank mentionned in the stop. Its value must stay between 1 and the number of logical keys of the manual.
If this explanation doesn't help, could you share the content of your Stop001 object and the Manual999 object to which it belongs ?

**Reply by mveccia** [reply]
[Stop001]
Name=Corno 4
FirstAccessiblePipeLogicalKeyNumber=1		
NumberOfAccessiblePipes=30
NumberOfRanks=1
Displayed=N
Rank001=001
Rank001FirstPipeNumber=1
Rank001FirstAccessibleKeyNumber=1
Rank001PipeCount=30
Function=And
SwitchCount=1
Switch001=001

[Switch001]
Name=Corno 4
DefaultToEngaged=N
GCState=0
StoreInDivisional=Y
StoreInGeneral=Y
Displayed=Y
DispLabelText=
PositionX=1442
PositionY=200
ImageOn=Dati\Pedaliera\Corno 4\StopOn.png
ImageOff=Dati\Pedaliera\Corno 4\StopOff.png
MouseRadius=0
MouseRectLeft=0
MouseRectWidth=64
MouseRectTop=0
MouseRectHeight=64

[Rank001]
Name=Corno 4
NumberOfLogicalPipes=30
FirstMidiNoteNumber=1
WindchestGroup=1
Percussive=N

Pipe001Gain=0
Pipe001HarmonicNumber=4
Pipe001TrackerDelay=1
Pipe001IsTremulant=0
Pipe001=Dati\Pedaliera\Corno 4\A0\036-c.wav
Pipe001LoadRelease=N
Pipe001PitchTuning=0.00000
Pipe001ReleaseCount=3
Pipe001Release001=Dati\Pedaliera\Corno 4\R0\036-c.wav
Pipe001Release001MaxKeyPressTime=-1
Pipe001Release001IsTremulant=0

Pipe001Release002=Dati\Pedaliera\Corno 4\R1\036-c.wav
Pipe001Release002MaxKeyPressTime=200
Pipe001Release002IsTremulant=0

Pipe001Release003=Dati\Pedaliera\Corno 4\R2\036-c.wav
Pipe001Release003MaxKeyPressTime=300
Pipe001Release003IsTremulant=0
.....
[WindchestGroup001]
Name=Pedaliera
NumberOfEnclosures=0
NumberOfTremulants=0
;Tremulant001=001

[Manual000]
Name=Pedaliera
NumberOfLogicalKeys=30
FirstAccessibleKeyLogicalKeyNumber=1
FirstAccessibleKeyMIDINoteNumber=36
NumberOfAccessibleKeys=30
DisplayKeys=30
MIDIInputNumber=1

; ----------------------------------------------------------------------do
Key001ImageOn=Dati\Pedaliera\biancoOn.png
Key001ImageOff=Dati\Pedaliera\biancoOff.png
Key001Offset=0
Key001Width=0
Key001MouseRectLeft=0
Key001MouseRectTop=0
Key001MouseRectWidth=19
Key001MouseRectHeight=216
....
NumberOfCouplers=2
Coupler001=001
Coupler002=002

NumberOfStops=3
Stop001=001
Stop002=002
Stop003=003

NumberOfSwitches=5
Switch001=001
Switch002=002
Switch003=003
Switch004=010
Switch005=011

Displayed=Y
PositionX=650
PositionY=785
DisplayFirstNote=36

**Reply by eturpault** [reply]
Looking at the lines that you provided I see no reason for this warning.
Looking at the GrandOrgue source code, this warning is raised when the function called parseLong returns False. This happens if the value after the = character has a null length or it doesn't contain digits between 0 and 9 or +/- characters at the beginning. It is not your case with the attribute FirstAccessiblePipeLogicalKeyNumber which looks to be correctly =1.
If you open your .organ file with an editor and you enable the hidden characters display, do you have maybe a special character after the 1 ?

**Reply by mveccia** [reply]
![Caratteri](https://user-images.githubusercontent.com/46837801/211723556-8468ec26-0573-4c8b-a7cb-1913f5cb0f81.jpg)
I don't know if you can see the picture, but apparently there are no hidden characters. the data "1" is retrieved from a program I created where the data of a pipe organ are stored in a mysql database. the 1 in question is stored in an INT field neither more nor less than the other pipe organ data, such as the 36 of the following line, and is inserted in the line in the same way as all the other data. I'll try to change the charset, but I don't think that's the cause...

**Reply by eturpault** [reply]
Yes the picture is visible, it doesn't show the attribute FirstAccessiblePipeLogicalKeyNumber of a Stop section, but I suppose that it is like in the picture for other attributes with no hidden character. I don't understand why GO detects a wrong integer value for this attribute only and for all stops, maybe someone else in this forum will have an idea.

**Reply by mveccia** [reply]
Dear friend, you must excuse me but you were right... I sent you the image relating to the manual forgetting that the error was on the stop, where actually there was one tabulator too many! I cleaned the code now I have no more error messages.

**Reply by eturpault** [reply]
I am glad to know that it has solved your problem !

**Reply by mveccia** [reply]
Do you know where I can find sample code for a crescendo pedal?

**Reply by eturpault** [reply]
Do you mean code to put in the .organ file ? GrandOrgue has already a built-in panel permitting to manage a crescendo pedal. In the .organ file, the usage of dedicated setter elements should permit to implement a crescendo pedal (it is explained in the GO documentation).
Concerning a sample code, I found [this organ](https://sites.google.com/view/orgues-electroniques/905-v8-orgue-conn-653?authuser=0) which has a crescendo pedal (it can be downloaded at the end of the page). It uses Divisional objects to implement it.
